src/sentry/workflow_engine/migration_helpers/utils.py

```diff
@@ -20,13 +22,36 @@ def get_action_description(action: AlertRuleTriggerAction) -> str:
     """
 
     if action.type == AlertRuleTriggerAction.Type.EMAIL.value:
-        if action.target:
-            if action.target_type == AlertRuleTriggerAction.TargetType.USER.value:
-                action_target_user = cast(RpcUser, action.target)
-                return "Email " + action_target_user.email
-            elif action.target_type == AlertRuleTriggerAction.TargetType.TEAM.value:
-                action_target_team = cast(Team, action.target)
-                return "Email #" + action_target_team.slug
+        if action.target_type == AlertRuleTriggerAction.TargetType.USER.value:
+            missing_user_text = "Email [missing user]"
+            try:
+                org_member = OrganizationMember.objects.get(
+                    user_id=action.target_identifier,
+                    organization=action.alert_rule_trigger.alert_rule.organization,
+                )
```

Comment:
I'm not sure we can use the email from here, based on this comment:

https://github.com/getsentry/sentry/blob/a9742b0192f372a86beee206d24a0ab79870dad3/src/sentry/models/organizationmember.py#L205-L207

I have a feeling this column might go away, based on work @mifu67 has been doing to separate invites from org member https://github.com/getsentry/sentry/blob/41a70d5d8736f6a60502b518040d0362c35de792/src/sentry/models/organizationmemberinvite.py#L72

Reply:
I am using [user_email](https://github.com/getsentry/sentry/blob/master/src/sentry/models/organizationmember.py#L240-L242) which looks a little more stable, but I'll see if I can find another way to accomplish this too.

Reply:
Ahh, I missed that... the comment there doesn't inspire a lot of confidence.

I wonder if it's ok to use https://github.com/getsentry/sentry/blob/a9742b0192f372a86beee206d24a0ab79870dad3/src/sentry/models/organizationmember.py#L450-L462

I think https://github.com/getsentry/sentry/blob/a9742b0192f372a86beee206d24a0ab79870dad3/src/sentry/models/organizationmember.py#L458 avoids that transaction error you were getting in tests

Reply:
Also wondering if we should update target to use this instead. There will be other cases where it gets called inside of a transaction

Reply:
I was thinking about updating that as well initially but was worried about having to track down all the places it's referenced

Reply:
Another option is to add this decorator into the call inside of the action. I'm not sure what the downsides are to doing that though

Reply:
Will this logic be used within any of our migrations by chance? If so I'd highly recommend against using this decorator.

Just for context, using this decorator is a massive code smell as it opens the door for data inconsistency or transaction timeouts, which is why we fail on this condition in the first place.

Reply:
Yes this is used during dual write because we needed a way to come up with a meaningful name for the workflow instances